Eventnet/Eventnet.Api/Services/SaveServices/ConsumeEventService.cs

```diff
@@ -4,45 +4,39 @@
 using RabbitMQ.Client;
 using RabbitMQ.Client.Events;
 
-namespace Eventnet.Services;
+namespace Eventnet.Services.SaveServices;
 
-public class ConsumeEventService : BackgroundService
+public class RabbitMqConsumeEventService : IConsumeEventService
 {
     private readonly IModel channel;
     private readonly IConnection connection;
     private readonly string queue;
-    private readonly IRabbitMqMessageHandler rabbitMqMessageHandler;
-
-    public ConsumeEventService(IRabbitMqMessageHandler rabbitMqMessageHandler, RabbitMqConfig config)
+    
+    public RabbitMqConsumeEventService(RabbitMqConfig config)
     {
-        this.rabbitMqMessageHandler = rabbitMqMessageHandler;
         queue = config.Queue;
         var connectionFactory = new ConnectionFactory { HostName = config.HostName };
         connection = connectionFactory.CreateConnection();
         channel = connection.CreateModel();
     }
 
-    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    public void ConsumeAndHandle(Action<RabbitMqMessage> handle)
     {
-        stoppingToken.ThrowIfCancellationRequested();
         var consumer = new EventingBasicConsumer(channel);
         consumer.Received += (ch, ea) =>
```

Comment:
```suggestion
        consumer.Received += (_, ea) =>
```
  <Eventnet.Api>\Services\SaveServices\ConsumeEventService.cs:813 Parameter 'ch' is never used